### ISSUE TITLE
Fetching lower properties from  tree

### ISSUE THREAD

**Issue by mmeroberts** [stathissideris/spectacles#4]
I am trying to use the lens to navigate yaml sructures changed into maps. One of the features of this is that some layers of the yaml are just names that contain maps.  Is there a way to skip a layer in the get-in function to say for eaxmple (lens/get-in assembly [::target :feature * :value])  to get all the values of the features no matter what their name is?

**Comment by stathissideris** [comment]
I'm not sure I understand, could you please post a specific example of the structure you are trying to navigate?

**Comment by mmeroberts** [comment]
Sorry - here is a detailed example:
<pre>
{:parent "fred"
 :children { :bert  { :age 1 } 
             :sam   { :age 3 }
             :fiona { :age 6 }
             }
    }
</pre>

<pre>
(s/def ::parent string?)
(s/def ::children (s/map-of keyword? ::child))
(s/def ::child (s/keys :req-un [ ::age ]))
(d/def ::age number?)
</pre>

You will notice the use of (s/map-of keyword? ...) to overcome the fact that we do not know the names.

This is not how you would design a Clojure map but it is very common in yaml.  The keys in the :children map are the issue.  What I would like to be able to access all the children without knowing their names.

**Comment by stathissideris** [comment]
Ok, that's clear. This is partly supported right now, but in order to do this you need a spec for your top level map that looks like this:

```clojure
(s/def ::top-map (s/keys :req-un [::parent ::children]))
```
You can then do:

```clojure
> (lens/get-in m [::top-map :children :bert])
{:age 1}
```

And in this case spectacles actually checks that the key that you provide is indeed valid. If you get confused and use a string instead of a keyword you get an error:

```clojure
> (lens/get-in m [::top-map :children "sam"])
ExceptionInfo Invalid key "sam" for spec :foo/children (valid keys: #function[clojure.core/keyword?])  clojure.core/ex-info (core.clj:4617)
```

But unfortunately it does get confused if you try and go deeper, which is a bug, so thanks for tracking this down! This should work but it doesn't:

```clojure
> (lens/get-in m [::top-map :children :bert :age])
IllegalArgumentException No implementation of method: :specize* of protocol: #'clojure.spec.alpha/Specize found for class: nil  clojure.core/-cache-protocol-fn (core_deftype.clj:568)
```

**Comment by mmeroberts** [comment]
Thanks - however what if I did not know the actual names? but I wanted to get the ages?  Without knowing the name I can't fetch them.  In specter there is a defined wild card that would navigate all the chidrena and fetch the ages.
e.g:
<code>
(lens/get-in m [::top-map :children :all :age])
</code>

**Comment by stathissideris** [comment]
Sorry, but this is out of the scope of this library, I don't intend for spectacles to have feature parity with specter. The aim is to cover simpler cases (such as the `get-in` equivalent) and check them with spec.

**Comment by mmeroberts** [comment]
I am sorry about that - but surely this use case from very common yaml is always going to be an issue and Spec allows for unnamed values in the s/map-of and using spec which is understandable rather than specter which is too complex in my mind would be much better.

**Comment by PabloReszczynski** [comment]
I am running into the same issue handling JSON maps, where map keys are not keywords but strings or integers.

**Comment by stathissideris** [comment]
This is now fixed in 0.3.4 (no spectre-like functionality was added, just the bug mentioned above).
